pkg/sync/sync_service.go

```diff
@@ -121,7 +121,7 @@ func (syncService *SyncService[H]) initStoreAndStartSyncer(ctx context.Context,
 	if initial.IsZero() {
 		return errors.New("failed to initialize the store and start syncer")
 	}
-	if err := syncService.store.Init(ctx, initial); err != nil {
+	if err := syncService.store.Append(ctx, initial); err != nil {
```

Comment:
still need to debug this, this is causing issues.

Reply:
i was running into the same thing here https://github.com/evstack/ev-node/pull/2606. havent had a moment to get back to it